CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -399,15 +399,8 @@ public static function checkFinancialTypeHasDeferred($params, $contributionID =
       $financialTypeID = $params['prevContribution']->financial_type_id;
     }
     if (($contributionID || !empty($params['price_set_id'])) && empty($lineItems)) {
-      if (!$contributionID) {
-        CRM_Price_BAO_PriceSet::processAmount($priceSetFields,
-          $params, $items);
-      }
-      else {
-        $items = CRM_Price_BAO_LineItem::getLineItems($contributionID, 'contribution', TRUE, TRUE, TRUE);
-      }
-      if (!empty($items)) {
-        $lineItems[] = $items;
+      if (!empty($orderLineItems)) {
```

Comment:
Do we need the if (!empty ) here - I feel like it has carried over from the original but I suspect it is no longer needed

Reply:
If I change the function signature to default [] instead of null then I think that would be ok, since then the loop won't choke iterating over null.

Reply:
yeah - it looks like the only 'live' call to this would always pass an array but the test calls don't